builder/views.py

```diff
@@ -118,13 +118,22 @@ def get(self, request: Request, key: str, **kwargs) -> HttpResponse:
 
         return JsonResponse(self._get(request, course))
 
+    @login_required_method(redirect_url=None)
+    def get(self, request: Request, remote_id: int, **kwargs) -> HttpResponse:
+        course = get_object_or_404(Course, remote_id=remote_id)
+
+        if not course.has_access(request, Permission.READ):
+            return JsonResponse({"success": False, "error": f"No access to instance {course.remote_id}"})
+
+        return JsonResponse(self._get(request, course))
+
     @login_required_method(redirect_url=None)
     def post(self, request: Request, key: str, **kwargs) -> HttpResponse:
```

Comment:
Please add id access support for this as well. It doesn't make sense for post to not support it when get and put do.